src/bun.js/api/bun/socket.zig

```diff
@@ -1651,6 +1674,11 @@ fn NewSocket(comptime ssl: bool) type {
             if (!this.socket.isClosed()) {
                 this.socket.close(0, null);
             }
+            if (this.connection != null) {
```

Comment:
nitpick:

```
```suggestion
            if (this.connection) |connection| {
                 this.connection = null;
                 connection.deinit();
            }
```